code_relocation: Add NOCOPY feature breaks windows builds

**Describe the bug**
PR #42859 Breaks Windows builds on certain platforms

The PR modified gen_relocate_app.py to add COPY/NOCOPY tag at the end of a string that was being constructed. The code then does a line.split() using the ":" character which works for Linux but for Windows it doesn't since the path to the file has a "<drive>:". This previously worked because the line split was only doing two items but now with three it splits after the c:.

**To Reproduce**
Steps to reproduce the behavior under Windows:
1. west build -b mimxrt595_evk_cm33 -p always samples\hello_world

**Expected behavior**
Build without errors.

**Impact**
Project does not build for certain boards in Windows (at least the mimxrt595_evk_cm33)

**Logs and console output**
Build log with some embedded debug print statements preceded by DJLZORK:

```
[147/163] Generating code_relocation.c, include/generated/linker_relocate.ld
C:\work\zephyrproject\zephyr\scripts\gen_relocate_app.py:426: UserWarning: File: C Not found
  warnings.warn("File: "+file_name+" Not found")
DJLZORK:  SRAM:C:/work/zephyrproject/zephyr/soc/arm/nxp_imx/rt5xx/flash_clock_setup.c:COPY
DJLZORK:  SRAM C /work/zephyrproject/zephyr/soc/arm/nxp_imx/rt5xx/flash_clock_setup.c:COPY
[148/163] Building C object zephyr/CMakeFiles/code_relocation_source_lib.dir/code_relocation.c.obj
zephyr/code_relocation.c: In function 'data_copy_xip_relocation':
zephyr/code_relocation.c:10:1: warning: useless type name in empty declaration
   10 | void;
      | ^~~~
[153/163] Linking C executable zephyr\zephyr_pre0.elf
FAILED: zephyr/zephyr_pre0.elf zephyr/zephyr_pre0.map
```

**Environment (please complete the following information):**
 - OS: (Windows)
 - Toolchain (e.g Zephyr SDK, ...)
 -  a0f53f4da939

**Additional context**
Add any other context that could be relevant to your issue, such as pin setting,
target configuration, ...

--- a/cmake/modules/extensions.cmake
+++ b/cmake/modules/extensions.cmake
@@ -1273,7 +1273,7 @@ function(zephyr_code_relocate file location)
   endif()
   set_property(TARGET code_data_relocation_target
     APPEND PROPERTY COMPILE_DEFINITIONS
-    "${location}:${file}:${copy_flag}")
+    "${location}:${copy_flag}:${file}")
 endfunction()
 
 # Usage:

--- a/scripts/gen_relocate_app.py
+++ b/scripts/gen_relocate_app.py
@@ -418,7 +418,7 @@ def create_dict_wrt_mem():
         if ':' not in line:
             continue
 
-        mem_region, file_name, copy_flag = line.split(':', 2)
+        mem_region, copy_flag, file_name = line.split(':', 2)
 
         file_name_list = glob.glob(file_name)
         if not file_name_list:

